src/test/ui/issues/issue-76077.rs

```diff
@@ -0,0 +1,10 @@
+pub mod foo {
+    pub struct Foo {
+        you_cant_use_this_field: bool,
+    }
+}
+
+fn main() {
+    foo::Foo {};
+    //~^ ERROR cannot construct `Foo` with struct literal syntax due to inaccessible fields
+}
```

Comment:
Can we add a test for the following case?

```rust
pub mod foo {
    pub struct Foo {
        you_cant_use_this_field: bool,
    }
    impl Foo {
        pub fn new() -> Foo {
            Foo {
                you_cant_use_this_field: true,
            }
        }
    }
}

fn main() {
    let foo::Foo {} = foo::Foo::new();
}
```

Reply:
Added a test for this and implemented these changes for patterns.